dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/IBedrockTextGenerationIOService.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Nodes;
+using Amazon.BedrockRuntime.Model;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Bedrock input-output service to build the request and response bodies as required by the given model.
+/// </summary>
+internal interface IBedrockTextGenerationIOService
+{
+    /// <summary>
+    /// Builds InvokeModelRequest Body parameter to be serialized. Object itself dependent on model request parameter requirements.
+    /// </summary>
+    /// <param name="modelId">The model ID to be used as a request parameter.</param>
+    /// <param name="prompt">The input prompt for text generation.</param>
+    /// <param name="executionSettings">Optional prompt execution settings.</param>
+    /// <returns>The invoke request body per model requirements for the InvokeAsync Bedrock runtime call.</returns>
+    internal object GetInvokeModelRequestBody(string modelId, string prompt, PromptExecutionSettings? executionSettings = null);
```

Comment:
Is object the right type to represent the request body?

Reply:
Each implementation has a different model for the request body that will be sent to a `PostJsonAsync`. I think this is fair implementation as is also internal.